taichi/program/state_flow_graph.h

```diff
@@ -24,33 +24,33 @@ class StateFlowGraph {
   struct Node {
     //  TODO: make use of IRHandle here
     IRNode *root;
-    std::string kernel_name;
+    std::string task_name;
+    // Incremental ID to identify the i-th launch of the task.
+    int launch_id;
+    // For |input_edges|, each state could map to exactly one node.
+    // For |output_edges|, each state could map to at least one node.
     Edges input_edges, output_edges;
-  };
-
-  StateToNodeMapping latest_state_owner;
-
-  std::vector<std::unique_ptr<Node>> nodes;
 
-  Node *initial_node;  // The initial node holds all the initial states.
+    std::string string() const;
+  };
 
-  StateFlowGraph() {
-    nodes.push_back(std::make_unique<Node>());
-    initial_node = nodes.back().get();
-    initial_node->kernel_name = "initial_state";
-  }
+  StateFlowGraph();
 
   void print_edges(const Edges &edges);
 
   void print();
 
-  void dump_dot(const std::string &fn) {
```

Comment:
BTW, what was `fn` intended for..?

Reply:
Maybe **f**ile **n**ame?

Reply:
Yeah, that's for filename. In some cases, we want to dump intermediate results that are not exposed to Python. E.g., before/after fusion :-) I think we are good for now.

Reply:
Thx!